### ISSUE TITLE
Unable to remove Node, when there are multiple

### ISSUE THREAD

**Issue by rodymolenaar** [google-ar/sceneform-android-sdk#331]
I'm unable to remove a node when there are multiple Nodes. I'm using this function to create a node which I add as a child to an AnchorNode which I add to my root node. (A class property) But on removeButton click, only the last created Node is removed. And all subsequent calls from other nodes return ``Remove/detach called on an anchor that was already removed/detached.``

Does anyone know how I can solve this or identify if it's a library issue? 

And big thanks to everyone actually creating the library 🤗 

```kotlin
val anchor = hitResult.createAnchor()
val anchorNode = AnchorNode(anchor)
anchorNode.setParent(root)

// Create the transformable car and add it to the anchor.
val car: TransformableNode = createScene(anchorNode)
car.setParent(anchorNode)
car.select()
```

```kotlin
 private fun createScene(anchorNode: AnchorNode): TransformableNode {

        val base = TransformableNode(arFragment.transformationSystem)
        base.renderable = carRenderable
        base.localPosition = Vector3(0.0f, 0.0f, 0.0f)

        val customCardRenderable = cardRenderable.makeCopy()

        val removeButton = customCardRenderable.view.findViewById<Button>(R.id.remove_button)

        removeButton.setOnClickListener {
            Log.i("AR", "Attempt to remove" + anchorNode.name)
            root.removeChild(anchorNode)
            anchorNode.anchor.detach()
        }

        val cardNode = Node()
        cardNode.renderable = customCardRenderable
        cardNode.localPosition = Vector3(0.0f, 1f, 0.0f)

        base.addChild(cardNode)

        return base
    }
```

Note: I've also tried calling setParent(null) both on `anchorNode` and `base`.

**Comment by rodymolenaar** [comment]
Might be related to #330

**Comment by dsternfeld7** [comment]
Calling setOnClickListener will override the previously set click listener, so only the most recently set click listener is getting called.

**Comment by rodymolenaar** [comment]
Thanks for the heads up. Closing as https://github.com/google-ar/sceneform-android-sdk/issues/330#issuecomment-423591958 contains my answer
